src/uu/arch/src/arch.rs

```diff
@@ -12,16 +12,18 @@ extern crate uucore;
 use platform_info::*;
 
 use clap::{crate_version, App};
+use uucore::error::{FromIo, UResult};
 
 static ABOUT: &str = "Display machine architecture";
 static SUMMARY: &str = "Determine architecture name for current machine.";
 
-pub fn uumain(args: impl uucore::Args) -> i32 {
+#[uucore_procs::gen_uumain]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     uu_app().get_matches_from(args);
 
-    let uts = return_if_err!(1, PlatformInfo::new());
+    let uts = PlatformInfo::new().map_err_context(|| "arch: ".to_string())?;
```

Comment:
I think the correct context here would be "cannot get system name". I don't know how to make `arch` fail to check this, but I looked at [GNU's source code](https://github.com/coreutils/coreutils/blob/da0d448bca62c6305fc432f67e2c5ccc2da75346/src/uname.c#L287) and that's what they seem to be printing.
In any case you don't have to add "arch: " as the context, because the utlity name is always added, so right now in case of failure it would be printed twice (like `arch: arch: ....`).

Reply:
I would like to change this to follow GNU's behavior. Should I add commit to this branch or open a new PR?
Thanks for reviewing!

Reply:
Please open a new PR :)